src/App.tsx

```diff
@@ -1,27 +1,56 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleGetAll = () => {
+    getAll().then(setVisibleGoods);
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleGet5First = async () => {
+    const goods = await get5First();
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    return setVisibleGoods(goods);
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const handleGetRedGoods = () => {
+    getRedGoods().then(setVisibleGoods);
+  };
```

Comment:
Use either await or promises. Mixing the two for the same task is not a good style